cotel/src/main/scala/day5/Trampolines.scala

```diff
@@ -0,0 +1,25 @@
+package day5
+
+import scala.annotation.tailrec
+import scala.io.Source
+
+object Trampolines {
+
+  @tailrec def instructionsUntilOut(instructions: List[Int], i: Int = 0, steps: Int = 0)(f: Int => Int): Int = {
```

Comment:
that's the key, defaults values for not create a secondary method which initialize the variables with 0

Reply:
Preferably this is the way to go, unless the input needs some preparation in which case I make a internal function for the recursion and then I transform the input and call the recursion.

You could directly accept the tranformed input as input, but these methods has no parent above them who can prepare the calls so 🤷‍♂️